ipld/tests/ipld-traversal-vectors/selector_explore.json

```diff
@@ -11,7 +11,7 @@
         "explore": [
             {
                 "ipld": null,
-                "path_segment": 0
+                "path_segment": "0"
```

Comment:
Seems like we had a breaking and suboptimal change (representing numbers in with strings). Are you sure we want to carry on with this?

Reply:
We previously allowed both strings and integers and then converted them on the fly (both from string to integer and integer to string). Disallowing integer values doesn't make it any less or more typed than before, imo.

Reply:
Why do we want to disallow them? Wouldn't it be less flexible? I could imagine a client that "optimizes" its calls a bit that would then need to change.

Reply:
They're disallowed in `libipld`. We have to roll our own code to allow integers as path segments. Given that it doesn't actually buy us any additional type safety (I'm ready to be shown wrong on this, btw), I don't think having a custom implementation of path segments is worth it.

Reply:
Ah, ok. If the upstream disallows them then all good. Perhaps someone will issue a special feature request in the future if this flexibility is needed.